File: windows/CodePushUtils.cs
using Newtonsoft.Json.Linq;
using System;
using System.Diagnostics;
using System.Threading.Tasks;
using Windows.Storage;
using Windows.Storage.Streams;
using Windows.System.Profile;

namespace CodePush.ReactNative
{
    internal class CodePushUtils
    {
        internal async static Task<JObject> GetJObjectFromFileAsync(StorageFile file)
        {
            string jsonString = await FileIO.ReadTextAsync(file).AsTask().ConfigureAwait(false);
            if (jsonString.Length == 0)
            {
                return new JObject();
            }

            try
            {
                return JObject.Parse(jsonString);
            }
            catch (Exception)
            {
                return null;
            }
        }

        internal static void Log(string message)
        {
            Debug.WriteLine("[CodePush] " + message, CodePushConstants.ReactNativeLogCategory);
        }

        internal static void LogBundleUrl(string path)
        {
            Log("Loading JS bundle from \"" + path + "\"");
        }

        internal static string GetDeviceId()
        {
            HardwareToken token = HardwareIdentification.GetPackageSpecificToken(null);
            IBuffer hardwareId = token.Id;
            var dataReader = DataReader.FromBuffer(hardwareId);

            var bytes = new byte[hardwareId.Length];
            dataReader.ReadBytes(bytes);

            return BitConverter.ToString(bytes);
        }
    }
}

File: windows/CodePush/CodePushUtils.cpp
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

#include "pch.h"
#include "CodePushUtils.h"

namespace Microsoft::CodePush::ReactNative
{
	using namespace winrt::Windows::Storage;

	/*static*/ void CodePushUtils::Log(winrt::hstring message)
	{
		OutputDebugStringW(L"[CodePush] ");
		OutputDebugStringW(message.c_str());
		OutputDebugStringW(L"\n");
	}

	/*static*/ void CodePushUtils::Log(const winrt::hresult_error& ex)
	{
		OutputDebugStringW(L"[CodePush] Exception ");
		OutputDebugStringW(ex.message().c_str());
		OutputDebugStringW(L"\n");
	}

	/*static*/ void CodePushUtils::LogBundleUrl(const IStorageFile& bundle)
	{
		CodePushUtils::Log(L"Loading JS bundle from \"" + bundle.Path() + L"\"");
	}
}
